receiver/sqlserverreceiver/scraper.go

```diff
@@ -470,3 +490,258 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
+	rows, err := s.client.QueryRows(ctx)
+	if err != nil {
+		if !errors.Is(err, sqlquery.ErrNullValueWarning) {
+			return plog.Logs{}, fmt.Errorf("sqlServerScraperHelper failed getting log rows: %w", err)
+		}
+		// TODO: ignore this for now.
+		s.logger.Warn("problems encountered getting log rows", zap.Error(err))
+	}
+
+	var errs []error
+	logs := plog.NewLogs()
+
+	resourceLog := logs.ResourceLogs().AppendEmpty()
+	resourceLog.Resource().Attributes().PutStr("db.system.type", "microsoft.sql_server")
+
+	scopedLog := resourceLog.ScopeLogs().AppendEmpty()
+	scopedLog.Scope().SetName("github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver")
+	scopedLog.Scope().SetVersion("v0.0.1")
+	for _, row := range rows {
+		queryHashVal := hex.EncodeToString([]byte(row[queryHash]))
+		queryPlanHashVal := hex.EncodeToString([]byte(row[queryPlanHash]))
+		contextInfoVal := hex.EncodeToString([]byte(row[contextInfo]))
+
+		// clientPort could be null, and it will be converted to empty string with ISNULL in our query. when it is
+		// an empty string, clientPortNumber would be 0.
+		clientPortNumber := 0
+		if row[clientPort] != "" {
+			clientPortNumber, err = strconv.Atoi(row[clientPort])
+			if err != nil {
+				s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing client port number. original value: %s, err: %s", row[clientPort], err))
+			}
+		}
+
+		sessionIDNumber, err := strconv.Atoi(row[sessionID])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing session id number. original value: %s, err: %s", row[sessionID], err))
+		}
+		blockingSessionIDNumber, err := strconv.Atoi(row[blockingSessionID])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing blocking session id number. value: %s, err: %s", row[blockingSessionID], err))
+		}
+		waitTimeVal, err := strconv.Atoi(row[waitTime])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing wait time number. original value: %s, err: %s", row[waitTime], err))
+		}
+		openTransactionCountVal, err := strconv.Atoi(row[openTransactionCount])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing open transaction count. original value: %s, err: %s", row[openTransactionCount], err))
+		}
+		transactionIDVal, err := strconv.Atoi(row[transactionID])
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing transaction id number. original value: %s, err: %s", row[transactionID], err))
+		}
+		// percent complete and estimated completion time is a real value in mssql
+		percentCompleteVal, err := strconv.ParseFloat(row[percentComplete], 32)
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("sqlServerScraperHelper failed parsing percent complete. original value: %s, err: %s", row[percentComplete], err))
+		}
+		estimatedCompletionTimeVal, err := strconv.ParseFloat(row[estimatedCompletionTime], 32)
```

Comment:
On the documentation, this is bigint, not a float. https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-exec-requests-transact-sql?view=sql-server-ver16

Reply:
updated